web/client-api/src/main/java/io/deephaven/web/client/api/WorkerConnection.java

```diff
@@ -479,54 +481,56 @@ private Promise<Void> authUpdate() {
-            handshake.onHeaders(headers -> {
-                // unchecked cast is required here due to "aliasing" in ts/webpack resulting in BrowserHeaders !=
-                // Metadata
-                JsArray<String> authorization = Js.<BrowserHeaders>uncheckedCast(headers).get(FLIGHT_AUTH_HEADER_NAME);
-                if (authorization.length > 0) {
-                    JsArray<String> existing = metadata().get(FLIGHT_AUTH_HEADER_NAME);
-                    if (!existing.getAt(0).equals(authorization.getAt(0))) {
-                        // use this new token
-                        metadata().set(FLIGHT_AUTH_HEADER_NAME, authorization);
-                        CustomEventInit<JsRefreshToken> init = CustomEventInit.create();
-                        init.setDetail(new JsRefreshToken(authorization.getAt(0), sessionTimeoutMs));
-                        info.fireEvent(EVENT_REFRESH_TOKEN_UPDATED, init);
+        return UnaryWithHeaders.<ConfigurationConstantsRequest, ConfigurationConstantsResponse>call(
+                this, ConfigService.GetConfigurationConstants, new ConfigurationConstantsRequest())
+                .then(result -> {
+                    BrowserHeaders headers = result.getHeaders();
+                    // unchecked cast is required here due to "aliasing" in ts/webpack resulting in BrowserHeaders !=
+                    // Metadata
+                    JsArray<String> authorization =
+                            Js.<BrowserHeaders>uncheckedCast(headers).get(FLIGHT_AUTH_HEADER_NAME);
+                    if (authorization.length > 0) {
+                        JsArray<String> existing = metadata().get(FLIGHT_AUTH_HEADER_NAME);
+                        if (!existing.getAt(0).equals(authorization.getAt(0))) {
+                            // use this new token
+                            metadata().set(FLIGHT_AUTH_HEADER_NAME, authorization);
+                            CustomEventInit<JsRefreshToken> init = CustomEventInit.create();
+                            init.setDetail(new JsRefreshToken(authorization.getAt(0), sessionTimeoutMs));
+                            info.fireEvent(EVENT_REFRESH_TOKEN_UPDATED, init);
+                        }
                     }
-                }
-            });
-            handshake.onStatus(status -> {
-                if (status.isOk()) {
+
                     // schedule an update based on our currently configured delay
                     scheduledAuthUpdate = DomGlobal.setTimeout(ignore -> {
                         authUpdate();
                     }, sessionTimeoutMs / 2);
 
-                    resolve.onInvoke((Void) null);
-                } else {
-                    if (status.getCode() == Code.Unauthenticated) {
+                    result.getMessage().getConfigValuesMap().forEach((item, key) -> {
+                        if (key.equals("http.session.durationMs")) {
+                            sessionTimeoutMs = Double.parseDouble(item.getStringValue());
```

Comment:
We probably need to move setting the `sessionTimeoutMs` prior to using it in the `setTimeout` a few lines above?